great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -295,27 +343,250 @@ def run(
+                    )
+                )
+            else:
+                effective_parameter_builder_configs.append(parameter_builder_config)
+
+        return effective_parameter_builder_configs
+
+    @staticmethod
+    def _reconcile_parameter_builder_config(
+        parameter_builder: ParameterBuilder,
+        parameter_builder_config: dict,
+    ) -> dict:
+        effective_parameter_builder_config: dict = {}
+        batch_request: Optional[
+            Union[BatchRequest, RuntimeBatchRequest, dict]
+        ] = parameter_builder_config.pop("batch_request", None)
+        if batch_request is None:
+            batch_request = get_batch_request_as_dict(
+                batch_request=parameter_builder.batch_request
+            )
+
+        effective_parameter_builder_config["batch_request"] = batch_request
+
+        key: str
+        value: Any
+        current_value: Any
+        for key, value in parameter_builder_config:
+            if hasattr(parameter_builder, f"{key}"):
+                current_value = getattr(parameter_builder, key)
+                effective_parameter_builder_config[key] = value or current_value
+            else:
+                effective_parameter_builder_config[key] = value
+
+        return effective_parameter_builder_config
+
+    @staticmethod
+    def _reconcile_expectation_configuration_builder_configs_for_rule(
+        rule: Rule, rule_config: dict
+    ) -> List[dict]:
+        effective_expectation_configuration_builder_configs: List[dict] = []
+        expectation_configuration_builder_configs: Optional[
+            List[dict]
+        ] = rule_config.get("expectation_configuration_builders", [])
+
+        current_expectation_configuration_builders: Dict[
+            str, ExpectationConfigurationBuilder
+        ] = rule.expectation_configuration_builders
+        expectation_configuration_builder_config: dict
+        for (
+            expectation_configuration_builder_config
+        ) in expectation_configuration_builder_configs:
+            expectation_configuration_builder_name: str = (
+                expectation_configuration_builder_config["expectation_type"]
+            )
+            if (
+                expectation_configuration_builder_name
+                in current_expectation_configuration_builders
+            ):
+                expectation_configuration_builder: ExpectationConfigurationBuilder = (
+                    current_expectation_configuration_builders[
+                        expectation_configuration_builder_name
+                    ]
+                )
+                effective_expectation_configuration_builder_configs.append(
+                    RuleBasedProfiler._reconcile_expectation_configuration_builder(
+                        expectation_configuration_builder=expectation_configuration_builder,
+                        expectation_configuration_builder_config=expectation_configuration_builder_config,
+                    )
```

Comment:
largely personal preference but can we separate this into two lines? one to reconcile and the other to append? I'll leave this up to you but I feel like there's a lot going on here.

Reply:
@cdkini I also went back and forth on this one and decided lean towards brevity.

Reply:
that's fine by me. LGTM